peachjam/models/user_following.py

```diff
@@ -0,0 +1,255 @@
+        qs = CoreDocument.objects.all()
+        if self.last_alerted_at:
+            qs = qs.filter(created_at__gt=self.last_alerted_at)
+        if self.court:
+            qs = qs.filter(judgment__court=self.court)[:10]
+            return {
+                "followed_object": self.court,
+                "documents": qs,
+            }
+
+        elif self.author:
+            qs = qs.filter(genericdocument__author=self.author)[:10]
+            return {
+                "followed_object": self.author,
+                "documents": qs,
+            }
+        elif self.court_class:
+            qs = qs.filter(judgment__court__court_class=self.court_class)[:10]
+            return {
+                "followed_object": self.court_class,
+                "documents": qs,
+            }
+        elif self.court_registry:
+            qs = qs.filter(judgment__registry=self.court_registry)[:10]
+            return {
+                "followed_object": self.court_registry,
+                "documents": qs,
+            }
+        elif self.country:
+            qs = qs.filter(jurisdiction=self.country)[:10]
+            return {
+                "followed_object": self.country,
+                "documents": qs,
+            }
+        elif self.locality:
+            qs = qs.filter(locality=self.locality)[:10]
+            return {
+                "followed_object": self.locality,
+                "documents": qs,
+            }
+        elif self.taxonomy_topic:
+            topics = [self.taxonomy_topic] + [
+                t for t in self.taxonomy_topic.get_descendants()
+            ]
+            qs = qs.filter(taxonomies__topic__in=topics)[:10]
+            return {
+                "followed_object": self.taxonomy_topic,
+                "documents": qs,
+            }
+
+    @classmethod
+    def update_and_alert(cls):
+        users = get_user_model().objects.filter(following__isnull=False).distinct()
+        for user in users:
+            follows = UserFollowing.objects.filter(user=user)
+            documents = []
+            for follow in follows:
+                new = follow.get_new_followed_documents()
+                if new["documents"]:
+                    follow.last_alerted_at = timezone.now()
+                    follow.save()
+                    documents.append(new)
+            if documents:
+                cls.send_alert(user, documents)
+
+    @classmethod
+    def send_alert(cls, user, documents):
+        documents = documents
+        context = {
+            "followed_documents": documents,
+            "user": user,
+            "site": Site.objects.get_current(),
+        }
+        html = render_to_string(
+            "peachjam/emails/user_following_alert_email.html", context
+        )
+        plain_txt = render_to_string(
+            "peachjam/emails/user_following_alert_email.txt", context
+        )
+        subject = settings.EMAIL_SUBJECT_PREFIX + _("New documents have been published")
```

Comment:
how do we handle translations for other alerts? because we don't have a request here to determine the correct language to use (I know we chatted about this before, I don't recall the outcome). Perhaps now is the time to have a "preferred language" setting on the user object?

Reply:
added the preferred language to the user profile model, the override context allows us to set the language while rendering the email. also added the preferred language to the queryset.